docs/guide/sending-email.md

```diff
@@ -0,0 +1,280 @@
+
+### Example usage
+
+<AposCodeBlock>
+
+```js
+module.exports = {
+  extend: '@apostrophecms/piece-type',
+  options: {
+    lable: 'article',
+    pluralLabel: 'articles'
+  },
+  ...,
+  handlers(self) {
+    return {
+      'afterSave': {
+        async sendEmail(req, piece) {
+          const options = {
+            from: 'admin@mysite.com',
+            to: 'editors@mysite.com',
+            subject: 'New Article added'
+          };
+          try {
+            await self.email(req, 'email.html', { data: { piece } }, options);
+          } catch (err) {
+            self.apos.util.error('email notification error: ', err);
+          }
+        }
+      }
+    }
+  }
+};
+```
+
+<template v-slot:caption>
+/modules/article/index.js
+</template>
+
+</AposCodeBlock>
+
+
+<AposCodeBlock>
+
+```django
+<h1>A new article has been added to the site</h1>
+<p>Here is the blurb</p>
+{% data.piece.title %}
+{% area data.piece, 'blurb'}
+```
+
+<template v-slot:caption>
+/modules/article/views/email.html
+</template>
+
+</AposCodeBlock>
+
+In this example, we are creating a custom piece type that implements an article. The `afterSave` server event property is being added to the `handlers()` function. This event is emitted any time this custom module saves a new article and runs the function that is being passed in as a value. This function, in turn, sends out an email to the site editors. The data coming from the piece is passed into the `email.html` template through the `data` argument to add the title and blurb for the editors to review.
+
+The `self.email()` method returns `info`. This can be used to determine if message handoff to the mailing service has been completed successfully. Note: this does not mean that message delivery will complete successfully. The email could still be rejected en route or by the receiving server.
+The returned data can also be used along with the `stream` transporter to ensure that the header and body of the email have been correctly constructed.
+
+### Stream example
+
+<AposCodeBlock>
+
+```js
+module.exports = {
+  options: {
+    nodemailer: {
+      // enables the usage of the `stream` transport
+      streamTransport: true
+    }
+  }
+};
+```
+
+<template v-slot:caption>
+/modules/@apostrophecms/email
+</template>
+</AposCodeBlock>
+
+<AposCodeBlock>
+
+```js
+module.exports = {
+  extend: '@apostrophecms/piece-type',
+  options: {
+    lable: 'article',
+    pluralLabel: 'articles'
+  },
+  ...,
+  handlers(self) {
+    return {
+      'afterSave': {
+        async sendEmail(req, piece) {
+          const options = {
+            from: 'admin@mysite.com',
+            to: 'editors@mysite.com',
+            subject: 'New Article added'
+          };
+          try {
+            const info = await self.email(req, 'email.html', { data: { piece } }, options);
+          }
+          console.log(info.envelope);
+          console.log(info.messgeId);
```

Comment:
Typo? Tested?

Reply:
Very much a typo - maybe VSC being "helpful". It autocorrects things oddly sometimes.